src/benchmarks/micro/libraries/System.Numerics.Vectors/Perf_VectorOfT.cs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Numerics.Tests
+{
+    public class Perf_VectorOfByte : Perf_VectorOf<byte> { }
+
+    public class Perf_VectorOfDouble : Perf_VectorOf<double> { }
+
+    public class Perf_VectorOfInt16 : Perf_VectorOf<short> { }
+
+    public class Perf_VectorOfInt32 : Perf_VectorOf<int> { }
+
+    public class Perf_VectorOfInt64 : Perf_VectorOf<long> { }
+
+    public class Perf_VectorOfSByte : Perf_VectorOf<sbyte> { }
+
+    public class Perf_VectorOfSingle : Perf_VectorOf<float> { }
+
+    public class Perf_VectorOfUInt16 : Perf_VectorOf<ushort> { }
+
+    public class Perf_VectorOfUInt32 : Perf_VectorOf<uint> { }
+
+    public class Perf_VectorOfUInt64 : Perf_VectorOf<ulong> { }
+
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD, Categories.JIT)]
+    public class Perf_VectorOf<T>
+        where T : struct
+    {
+        private static readonly Vector<T> Value1 = Vector<T>.One;
+
+        private static readonly Vector<T> Value2 = Vector<T>.One + Vector<T>.One;
+
+        private static readonly Vector<T> Value3 = Vector<T>.One + Vector<T>.One + Vector<T>.One;
```

Comment:
This is the same setup that the other intrinsified numerics tests use. The JIT should be recognizing and optimizing these patterns to be more efficient than a standard struct copy.

Reply:
I agree that it "should", but I would prefer to avoid relying on it. Moreover, we are from time to time using these benchmarks to compare against older .NET Core versions, Mono, and Full .NET Framework. It would be nice to be 100% sure that we are not benchmarking the static field access but just the actual vector work.

Reply:
Yes, however we also care that the JIT is doing the right thing in this scenario for newer runtimes. Having efficient codegen here is a requirement for several of our other vectorized algorithms in production code.